client/packages/common/src/hooks/useElectronClient/index.ts

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect } from 'react';
+
+const DISCOVERY_TIMEOUT = 5000;
+
+// Should match server/server/src/discovery.rs (FrontEndHost)
+export type FrontEndHost = {
+  protocol: 'http' | 'https';
+  port: number;
+  ip: string;
+  name: string;
+  clientVersion: string;
+  // This one is set by electron.ts
+  isLocal: boolean;
+};
+
+type ServerDiscovered = (event: object, value: FrontEndHost) => void;
+export type ElectronAPI = {
+  serverDiscovered: (eventInfo: ServerDiscovered) => void;
+  startServerDiscovery: () => void;
+  connectToServer: (server: FrontEndHost) => void;
```

Comment:
You mentioned that discovery is running in the background. Should it be stopped once the remote-server has been found?

Reply:
c77ef0ed6294b820bfec50c0a7ed782815eaf1f1